account_credit_control/models/credit_control_run.py

```diff
@@ -132,14 +132,15 @@ def _generate_credit_lines(self):
 
             if policy.do_nothing:
                 continue
-            (
-                policy_manual_lines,
-                policy_lines_generated,
-                policy_report,
-            ) = policy._generate_credit_lines(self, {"run_id": self.id})
-            manually_managed_lines |= policy_manual_lines
-            generated |= policy_lines_generated
-            report += policy_report
+            for company in self.company_id or self.env.companies:
```

Comment:
We shouldn't include in the same run several companies, as their records will be only visible if all companies are allowed at the same time.

Reply:
Well, right now the field `company_id` on the credit.control.run is not required. Users expects that if the field is null, it will create the credit lines for all the companies visible :thinking: 
We can make it required and forget about making this multicompany. Also we could make the field M2M in order to avoid doubts. What do you prefer?

Reply:
I think it should be done mandatory. There's no sense of executing a multi-company run that mixes up everything and to have to make the code aware of that. Imagine having the same partner with debts in several companies.

Unless it's your use case that you want to cover in one shot...

Reply:
The idea was to generate all the lines on the same execution

But, data is not mixed, communications and credit lines are created by company and partner, not only by partner, so they are not mixed.

Reply:
They all belong to the same run, which is the inconsistent thing.

Reply:
It is not if it has no company or multiple company associated

Reply:
Yes. The problem is the or party. Make required the field company_id